internal/plugin/convert/diagnostics.go

```diff
@@ -50,38 +40,57 @@ func AppendProtoDiag(diags []*proto.Diagnostic, d interface{}) []*proto.Diagnost
 	return diags
 }
 
-// ProtoToDiagnostics converts a list of proto.Diagnostics to a tf.Diagnostics.
-func ProtoToDiagnostics(ds []*proto.Diagnostic) tfdiags.Diagnostics {
-	var diags tfdiags.Diagnostics
+// ProtoToDiags converts a list of proto.Diagnostics to a diag.Diagnostics.
+func ProtoToDiags(ds []*proto.Diagnostic) diag.Diagnostics {
+	var diags diag.Diagnostics
 	for _, d := range ds {
-		var severity tfdiags.Severity
+		var severity diag.Severity
 
 		switch d.Severity {
 		case proto.Diagnostic_ERROR:
-			severity = tfdiags.Error
+			severity = diag.Error
 		case proto.Diagnostic_WARNING:
-			severity = tfdiags.Warning
+			severity = diag.Warning
 		}
 
-		var newDiag tfdiags.Diagnostic
-
-		// if there's an attribute path, we need to create a AttributeValue diagnostic
-		if d.Attribute != nil {
-			path := AttributePathToPath(d.Attribute)
-			newDiag = tfdiags.AttributeValue(severity, d.Summary, d.Detail, path)
-		} else {
-			newDiag = tfdiags.WholeContainingBody(severity, d.Summary, d.Detail)
-		}
-
-		diags = diags.Append(newDiag)
+		diags = append(diags, diag.Diagnostic{
+			Severity:      severity,
+			Summary:       d.Summary,
+			Detail:        d.Detail,
+			AttributePath: AttributePathToPath(d.Attribute),
+		})
 	}
 
 	return diags
 }
 
+func DiagsToProto(diags diag.Diagnostics) []*proto.Diagnostic {
+	var ds []*proto.Diagnostic
+	for _, d := range diags {
+		if err := d.Validate(); err != nil {
+			panic(fmt.Errorf("Invalid diagnostic: %s. This is always a bug in the provider implementation", err))
```

Comment:
why is this is an `Errorf` inside the panic? Should this func just return an error?

Reply:
I wanted to add the line that this should be considered a bug in the provider implementation. wanted the errors coming from the diagnostic validation to be terse